File: java/src/main/java/com/github/danhumphrey/thirtyfour/ui/table/StringCellParser.java
package com.github.danhumphrey.thirtyfour.ui.table;

import org.openqa.selenium.WebElement;

/**
 * Parses a string value from a table cell
 *
 */
public class StringCellParser implements CellParser<String> {

	/**
	 * Gets a string value from a cell
	 * @param WebElement cell the cell to extract the value from
	 * @return String the cell text
	 */
	@Override
	public String getValue(WebElement cell) {
		return cell.getText();
	}
}

File: dotnet/ThirtyFour.UI/Table/StringCellParser.cs
using OpenQA.Selenium;

namespace ThirtyFour.UI.Table
{
    /// <summary>
    /// Parses a string value from a table cell
    /// </summary>
    public class StringCellParser : ICellParser<string>
    {
        /// <summary>
        /// Gets a string value from a table cell
        /// </summary>
        /// <param name="cell">the cell to extract the value from</param>
        /// <returns>The cell text</returns>
        public string GetValue(IWebElement cell)
        {
            return cell.Text;
        }
    }
}
